prov/util/src/util_cntr.c

```diff
@@ -153,16 +153,27 @@ static int ofi_cntr_wait(struct fid_cntr *cntr_fid, uint64_t threshold, int time
 
 	assert(cntr->wait);
 
-	current_ms = fi_gettime_ms();
-	finish_ms = (timeout < 0) ? UINT64_MAX : current_ms + timeout;
-	for (; timeout < 0 || current_ms < finish_ms; current_ms = fi_gettime_ms()) {
-		timeout = timeout < 0 ? timeout : (int)(finish_ms - current_ms);
```

Comment:
The proposed changes look equivalent to simply changing the above line to:
```
timeout = timeout < 0 ? 0 : (int)(finish_ms - current_ms);
```
But that brings up the point that an infinite timeout will simply spin on cntr->progress.  I don't see that calling fi_wait() ends up doing anything.

However, if the caller provides a very large timeout (say 1000 years), progress ends up being stalled the same as if an infinite value were provided.

I think we need some other solution here.  It appears that this patch is working around an issue where the wait object is not directly associated with a counter, but some other resource.  Spinning on the progress loop in the infinite case only doesn't seem right.

Reply:
> Spinning on the progress loop in the infinite case only doesn't seem right.

Yes, I agree with you. It will just consume/waste CPU cycles instead of polling `epoll_wait` or mutex condition, etc.

The common issue here is how we can progress CQ/CNTR (w/o progress thread) to generate signaled events that we are waiting in the wait object.